Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/EcSupport.c

```diff
@@ -246,9 +312,50 @@ GetBoardId (
   }
   BoardInfo.Raw |= (UINT16) EcData;
 
-  *PlatformId = (UINT8) BoardInfo.AdlRvpFields.BoardId;
+  *BoardId = (UINT8)BoardInfo.AdlRvpFields.BoardId;
+}
+
+/**
+  Get the board ID from EC.
 
-  switch (BoardInfo.AdlRvpFields.BoardId) {
+  @param[out] PlatformId       Buffer to return the Platform ID based on EC BaordId
+
+**/
+VOID
+GetBoardId (
+  OUT UINT8                  *PlatformId
+)
+{
+  UINT8   BoardID;
+  FEATURES_CFG_DATA           *FeaturesCfgData;
+  //
+  // Set board ID to unknown initially
+  // in case there is any issue encountered
+  // below.
+  //
+  *PlatformId = 0xFF;
+  FeaturesCfgData = (FEATURES_CFG_DATA *) FindConfigDataByTag (CDATA_FEATURES_TAG);
+  if (FeaturesCfgData != NULL) {
+    DEBUG((DEBUG_INFO, "Features.ReadBoardID %x .....\n", FeaturesCfgData->Features.ReadBoardID));;
+    if (FeaturesCfgData->Features.ReadBoardID == 0){
+      GetBoardIdFromEC(&BoardID);
+    }
+    else if (FeaturesCfgData->Features.ReadBoardID == 1){
+      GetBoardIdFromSmbus(&BoardID);
+    }
+  }
+  else{
+    GetBoardIdFromEC(&BoardID);
+  }
```

Comment:
It would be more clear if using below logic to replace all the changes after line 337.  
FeaturesCfgData = (FEATURES_CFG_DATA *) FindConfigDataByTag (CDATA_FEATURES_TAG);
  if ((FeaturesCfgData != NULL) && (FeaturesCfgData->Features.ReadBoardID == 1)) {
    GetBoardIdFromSmbus(&BoardID);
   if (BoardID == BoardIdAdlNDdr5Crb){
      *PlatformId = PLATFORM_ID_ADL_N_DDR5_CRB;
   }
  return;
  }

Reply:
for this : 
if (BoardID == BoardIdAdlNDdr5Crb){
*PlatformId = PLATFORM_ID_ADL_N_DDR5_CRB;
}
return;

it need to use the preprocessor #ifdef PLATFORM_ADLN because the definition clashed with ADL-PS
#define BoardIdAdlNDdr5Crb                            0x0B
#define BoardIdAdlPSDdr5Rvp                           0x0B

The BoardID read form SmBus in ADL-N is 0xB so we have to use the preprocessor

Reply:
We also need to cater for the case if the FeatureCfgData pointer return NULL then it will read from EC by default.

Reply:
Since CFG data could not be used, we need refine this patch.
Maybe we could try the logic below:
  Check if EC is available firstly (), if yes, get boardID from EC, else get boardId from SMBUS.

Reply:
Updated patch without CfgData usage